packages/compass-components/src/components/guide-cue/guide-cue-groups.ts

```diff
@@ -0,0 +1,9 @@
+// When adding data to groups, make it readonly to make
+// GroupName type strict.
+// const GROUPS = [] as const; // of type {id: string, steps: number}
+const GROUPS: Array<{ id: string; steps: number }> = [];
+
+export const GROUP_STEPS_MAP = new Map<GroupName, number>();
+GROUPS.forEach(({ id, steps }) => GROUP_STEPS_MAP.set(id, steps));
```

Comment:
Just fyi: another way to do it, Map takes entries as an argument in constructor:

```suggestion
export const GROUP_STEPS_MAP = new Map<GroupName, number>(
  GROUPS.map(({ id, steps }) => [id, steps])
);
```

Reply:
oh thanks :)